### PR TITLE
Add builds for Eclipse 2018-12

### LINKED ISSUE #3300: Set up target for Eclipse 2018-12
With the imminent release of 2018-12, we should verify that tests etc run against 2018-12.  (I've been running locally with no issues).

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #3300 
- adds a 2018-12 target platform
- removes references to 32-bit platforms — has no material effect on what we build though
- updates README

### CODE REVIEW

File: eclipse/eclipse-2018-09/gcp-eclipse-2018-09.tpd [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
+    org.eclipse.epp.logging.aeri.feature.source.feature.group
+
+    org.eclipse.jetty.http
+    org.eclipse.jetty.servlet
+    org.eclipse.jetty.server
+    org.eclipse.jetty.util
 }
 
 // WTP SDKs aren't exposed through the main release links
 // (composite at http://download.eclipse.org/webtools/repository/photon)
```
Comment (chanseokoh) [inline eclipse/eclipse-2018-09/gcp-eclipse-2018-09.tpd]: Is it correct to have the "photon" reference here?

File: eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    org.eclipse.epp.logging.aeri.feature.source.feature.group
+
+    org.eclipse.jetty.http
+    org.eclipse.jetty.servlet
+    org.eclipse.jetty.server
+    org.eclipse.jetty.util
+}
+
+// WTP SDKs aren't exposed through the main release links
+// (composite at http://download.eclipse.org/webtools/repository/photon)
```
Comment (chanseokoh) [inline eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd]: ditto

File: eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ *      to update the corresponding .target file.
+ *    * Right-click in the editor and choose 'Set as Target Platform'
+ *      to update your IDE's target platform (regenerates the .target too)
+ */
+target "GCP for Eclipse 2018-12" with source requirements
+// Eclipse Platform 4.10
+// Eclipse WTP 3.12
+
+// Eclipse 2018-12 RC2 (composite at http://download.eclipse.org/releases/2018-12/)
+location "http://download.eclipse.org/staging/2018-12/" {
```
Comment (chanseokoh) [inline eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd]: "release" instead of "staging"? (Maybe not.)

**Comment by chanseokoh** [comment]
The 2018-12 builds are already failing. Injection error?

`org.eclipse.e4.core.di.InjectionException: Unable to process "LibraryClasspathContainerInitializer.resolverService": no actual value was found for the argument "ILibraryClasspathContainerResolverService".`

**Comment by chanseokoh** [comment]
Need to add below to `eclipse/.gitignore`.
```diff
 eclipse-2018-09/.project
 eclipse-2018-09/.settings
+eclipse-2018-12/.project
+eclipse-2018-12/.settings
```

File: eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd (hunk 3 above)
Comment (briandealwis) [reply eclipse/eclipse-2018-12/gcp-eclipse-2018-12.tpd]: I hesitated in putting `/releases` as the composite repository there is disabled for a couple of days before the official release.  But I think the actual repositories it points to are still available.  And `/staging` will disappear after the official release.  So I'll put in the full path the latest RC for 2018-12.

**Comment by chanseokoh** [comment]
I think you need to regenerate `gcp-eclipse-2018-09.target`.

In the `.tpd`,
```
// Eclipse 2018-09 (composite at http://download.eclipse.org/releases/2018-09/)
location "http://download.eclipse.org/releases/2018-09/201809191002/" {
```

but in `.target`
```
      <repository location="http://download.eclipse.org/releases/2018-09/201809071000/"/>
```

**Comment by briandealwis** [comment]
I can't think how that happened.  Good catch!

**Comment by briandealwis** [comment]
Oh dear:
```
1234767 [INFO] --- tycho-surefire-plugin:1.3.0:test (default-test) @ com.google.cloud.tools.eclipse.sdk.test ---
1235568 [INFO] {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, osgi.arch=x86_64}
1235569 [ERROR] Cannot resolve project dependencies:
1235570 [ERROR]   You requested to install 'org.eclipse.equinox.p2.iu; org.eclipse.equinox.ds 0.0.0' but it could not be found
1235570 [ERROR] 
1235571 [ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
```

It looks like `org.eclipse.equinox.ds` was removed as it was replaced by Felix's services implementation.  They had been merged a while back, and `org.eclipse.equinox.ds` was a proxy of sorts (require-bundle and re-exported).  With Tycho 1.3, the capabilities should pick it up I hope 🤞

**Comment by briandealwis** [comment]
Argh, missed the `verify-feature-completeness`. Sigh.

**Comment by briandealwis** [comment]
PTAL (and feel free to commit)

**Pull request merged**